src/actions/sources/newSources.js

```diff
@@ -208,8 +209,12 @@ export function newSource(source: Source) {
   };
 }
 
-export function newSources(sources: Source[]) {
+export function newSources(foundSources: Source[]) {
   return async ({ dispatch, getState }: ThunkArgs) => {
+    const sources = foundSources.filter(
+      source => !getSource(getState(), source.id)
+    );
```

Comment:
this is needed so that we don't clobber the original sources data

Reply:
This filtering looks like it will prevent us from being able to have multiple source actors for the same source.  We rely on being able to handle newSource notifications for the same Source, but with different contents for their source actors.  The sources reducer should merge the old and new information, instead of clobbering all the old state.